File: ComponentImpl/src/main/java/com/xiaojinzi/component/cache/DefaultCacheFactory.java
package com.xiaojinzi.component.cache;

import android.support.annotation.NonNull;

import com.xiaojinzi.component.Component;

public class DefaultCacheFactory implements Cache.Factory{

    private DefaultCacheFactory() {
    }

    public static final DefaultCacheFactory INSTANCE = new DefaultCacheFactory();

    @NonNull
    @Override
    public Cache build(CacheType type) {
        return new LruCache(type.calculateCacheSize(Component.getApplication()));
    }

}

File: ComponentImpl/src/main/java/com/xiaojinzi/component/cache/DefaultCacheFactory.kt
package com.xiaojinzi.component.cache;

import androidx.annotation.NonNull;

import com.xiaojinzi.component.Component;

public class DefaultCacheFactory<K, V> implements Cache.Factory<K, V>{

    public static final DefaultCacheFactory<Object, Object> INSTANCE = new DefaultCacheFactory();

    @NonNull
    @Override
    public Cache<K, V> build(CacheType type) {
        return new LruCache<K, V>(type.calculateCacheSize(Component.getApplication()));
    }

}
